### PR TITLE
Make contract size limit configurable

### PR DESCRIPTION
**Pull request by ganeshvanahalli**

### CODE REVIEW

File: params/config_arbitrum.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ type ArbitrumChainParams struct {
 	InitialArbOSVersion       uint64
 	InitialChainOwner         common.Address
 	GenesisBlockNum           uint64
+	MaxCodeSize               uint64 // Maximum bytecode to permit for a contract. 0 value implies params.MaxCodeSize
+	MaxInitCodeSize           uint64 // Maximum initcode to permit in a creation transaction and create instructions. o value implies params.MaxInitCodeSize
```
Comment (PlasmaPower) [inline params/config_arbitrum.go:33]: Let's add `json:"omitempty"` here as to not change the serialization of existing chain configs

File: params/config_arbitrum.go [hunk 2]
```diff
@@ -39,6 +41,20 @@ func (c *ChainConfig) IsArbitrumNitro(num *big.Int) bool {
...
+func (c *ChainConfig) MaxCodeSize() uint64 {
+	if c.ArbitrumChainParams.MaxCodeSize == 0 {
+		return MaxCodeSize
+	}
+	return c.ArbitrumChainParams.MaxCodeSize
+}
+
+func (c *ChainConfig) MaxInitCodeSize() uint64 {
+	if c.ArbitrumChainParams.MaxInitCodeSize == 0 {
+		return MaxInitCodeSize
```
Comment (PlasmaPower) [inline params/config_arbitrum.go:53]: Let's return `c.MaxCodeSize() * 2` here to make it a reasonable default even with a custom MaxCodeSize

**Review by PlasmaPower — CHANGES_REQUESTED** [review]
I have a couple minor comments on the chain config part but the use of the new parameter looks perfect

File: params/config_arbitrum.go [hunk 3]
```diff
@@ -29,8 +29,8 @@ type ArbitrumChainParams struct {
 	InitialArbOSVersion       uint64
 	InitialChainOwner         common.Address
 	GenesisBlockNum           uint64
-	MaxCodeSize               uint64 // Maximum bytecode to permit for a contract. 0 value implies params.MaxCodeSize
-	MaxInitCodeSize           uint64 // Maximum initcode to permit in a creation transaction and create instructions. o value implies params.MaxInitCodeSize
+	MaxCodeSize               uint64 `json:"maxCodeSize,omitempty,"`    // Maximum bytecode to permit for a contract. 0 value implies params.MaxCodeSize
+	MaxInitCodeSize           uint64 `json:"maxInitCodeSize,omitempty"` // Maximum initcode to permit in a creation transaction and create instructions. 0 value implies params.MaxInitCodeSize
```
Comment (PlasmaPower) [inline params/config_arbitrum.go:33]: Let's capitalize these JSON field names to match the others (they're capitalized as per the Go field name by default)

Comment (PlasmaPower) [reply params/config_arbitrum.go:33]: also, there's an extra comma on the first JSON attribute. I doubt it matters, but probably better not to have it

Comment (joshuacolvin0) [reply params/config_arbitrum.go:33]: should remove extra comma after `omitempty`

Comment (ganeshvanahalli) [reply params/config_arbitrum.go:33]: Cool! my bad for the typo

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**